test/e2e/end_user_test.go

```diff
@@ -51,4 +52,59 @@ var _ = Describe("Openshift on Azure end user e2e tests [EndUser]", func() {
 		// Expect(kerrors.IsForbidden(err)).To(Equal(true))
 		fmt.Printf("PDB create error: %v\n", err)
 	})
+
+	It("should deploy a template and check the visit counter increments", func() {
+		const TPL = "nodejs-mongodb-example"
+		var regex = regexp.MustCompile(`id="count-value">(\d+)<`)
+
+		// Create the template
+		template, err := c.tc.Templates("openshift").Get(
+			TPL, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Instantiate the template
+		_, err = c.tc.TemplateInstances(c.namespace).Create(
+			&templatev1.TemplateInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: c.namespace,
+				},
+				Spec: templatev1.TemplateInstanceSpec{
+					Template: *template,
+				},
+			})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Poll for the deployment status
+		err = wait.PollImmediate(2*time.Second, 5*time.Minute, c.templateInstanceIsReady)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Pull the route ingress from the namespace
+		route, err := c.rc.Routes(c.namespace).Get(TPL, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(route.Status.Ingress)).To(Equal(1))
+		host := route.Status.Ingress[0].Host
+		url := fmt.Sprintf("http://%s", host)
+
+		prevCounter := 0
+		currCounter := 0
+		httpc := &http.Client{
+			Timeout: 10 * time.Second,
+		}
+		// Hit the ingress 3 times, each time the hit counter should increment
+		for i := 0; i < 3; i++ {
+			resp, err := httpc.Get(url)
```

Comment:
I am not sure at this point the application is up and running so we may still want to retry 500s.